src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java

```diff
@@ -827,6 +829,10 @@ public void getUserProfile(String userId, AsyncCallback<UserProfile> callback) {
 		}
 	}
 
+	public FluentFuture<UserProfile> getMyUserProfile() {
+		return getUserProfile(null);
```

Comment:
We load the current user profile during app load, and it also validates the current access token.  
During Google SSO login the call is made servlet side, and only the access token is sent back to the client, so the response to this call is also used to find out if the user still needs to sign the Synapse terms of use